### PR TITLE
(CL) Swap tool promo

### PR DESCRIPTION
**Pull request by jonator**
## What is the purpose of the change

Adds a conditional promo drawer that animates above and behind the swap tool of one of the selected currencies is in _a_ CL pool. The button provides a link to the pools page with the concentrated pool filter in the URL params.

### ClickUp Task

Please provide a link to the ClickUp task for this pull request:

[ClickUp Task URL](https://app.clickup.com/t/8677q2nvp)

## Brief Changelog

Screenshot 2023-06-03 at 6 34 23 AM

- Adds animated background drawer to swap too iff one of the assets in the swap tool is contained with >=1 CL pool in the swap tool

## Key Workflows to test

- Select a CL asset, observe the drawer animate up
- Navigate to and from other pages, notice the drawer appear if the assets are still in a CL pool

## Testing and Verifying

Make sure the drawer only appears if a CL pool contains the selected asset.

### Figma
https://www.figma.com/file/RxNOvfvzmgtsgCPUetQvcG/Supercharged-Liquidity?type=design&node-id=1173-40832&t=gbUrjyXIcQuVLhgB-4

### CODE REVIEW

File: packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  pools: ObservableQueryPool[];
+  sendCurrency: AppCurrency;
+  outCurrency: AppCurrency;
+}> = ({ pools, sendCurrency, outCurrency }) => {
+  const router = useRouter();
+  const t = useTranslation();
+
+  // get the first denom to be found in CL pool
+  const denomInClPool = useMemo(() => {
+    const concentratedLiquidityPools = pools.filter(
```
Comment (MaxMillington) [inline packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx:24]: how big can the 'pools' array get? (just trying to get a sense of scale)

File: packages/web/components/swap-tool/index.tsx [hunk 2]
```diff
@@ -290,622 +296,618 @@ export const SwapTool: FunctionComponent<{
 
     return (
```
Comment (MaxMillington) [inline packages/web/components/swap-tool/index.tsx:297]: Is there any way this component can be broken into simpler parts? It's kind of long and hard to follow

File: packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx (hunk 1 above)
Comment (jonator) [reply packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx:24]: There's at most 1000 pools, but these pools get filtered down to ~200-300 routable pools that we use for swapping. 80% of pools have nearly no liquidity

File: packages/web/components/swap-tool/index.tsx (hunk 2 above)
Comment (jonator) [reply packages/web/components/swap-tool/index.tsx:297]: Yes I agree it's large but IMO breaking this up could be more confusing because then it's more difficult to track what the smaller components are used for. For sake of time, I'll leave it here and we can consider a refactor later?

Comment (MaxMillington) [reply packages/web/components/swap-tool/index.tsx:297]: SOunds good

File: packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx (hunk 1 above)
Comment (MaxMillington) [reply packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx:24]: That's fine then. If it was in the tens or hundreds of thousands then I think we might be able to make this function more efficient, but for 1300 that's def no problem

File: packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          onClick={() => {
+            router.push("/pools?pool=concentrated");
+          }}
+        >
+          <Image
+            alt="lightining bolt"
+            src="/icons/lightning.svg"
+            width={11}
+            height={22}
+          />
```
Comment (JoseRFelix) [inline packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx:73]: I think `lightning.svg` should be included in the `sprites.svg`
